File: packages/setpieces-r/setpieces/tests/testthat/test-events.R
test_that("extracts all set-piece kinds", {
  m <- sample_match()
  events <- extract_set_pieces(m)
  kinds <- sort(vapply(events, function(e) e$kind, character(1)))
  expect_equal(kinds, sort(c("corner", "corner", "corner", "free_kick", "throw_in", "penalty", "penalty")))
})

test_that("filters by kind", {
  m <- sample_match()
  corners <- extract_set_pieces(m, kinds = "corner")
  expect_length(corners, 3)
  expect_true(all(vapply(corners, function(e) e$kind == "corner", logical(1))))
})

test_that("corner fields are populated correctly", {
  m <- sample_match()
  corners <- extract_set_pieces(m, kinds = "corner")
  first <- corners[[1]]
  expect_equal(first$team_id, "t_A")
  expect_equal(first$player_name, "A. Corner Taker")
  expect_equal(first$subtype, "delivery")
  expect_equal(first$end_x, 88.0)
  expect_equal(first$end_y, 45.0)
})

test_that("penalty subtypes are goal/miss", {
  m <- sample_match()
  penalties <- extract_set_pieces(m, kinds = "penalty")
  subtypes <- sort(vapply(penalties, function(e) e$subtype, character(1)))
  expect_equal(subtypes, c("goal", "miss"))
})

test_that("event_index matches sorted_events", {
  m <- sample_match()
  events <- sorted_events(m)
  set_pieces <- extract_set_pieces(m)
  for (sp in set_pieces) {
    raw <- events[[sp$event_index]]
    expect_equal(raw$contestantId, sp$team_id)
  }
})

test_that("set_pieces_to_df builds a data.frame", {
  m <- sample_match()
  events <- extract_set_pieces(m)
  df <- set_pieces_to_df(events)
  expect_equal(nrow(df), length(events))
  expect_true(all(c("kind", "team_id", "end_x", "end_y") %in% names(df)))
})

File: packages/setpieces-py/tests/test_events.py
import setpieces as sp


def test_extract_all_kinds(match):
    events = sp.extract_set_pieces(match)
    kinds = sorted(e.kind for e in events)
    assert kinds == sorted([
        "corner", "corner", "corner", "free_kick", "throw_in", "penalty", "penalty",
    ])


def test_extract_filters_by_kind(match):
    corners = sp.extract_set_pieces(match, kinds=["corner"])
    assert len(corners) == 3
    assert all(e.kind == "corner" for e in corners)


def test_corner_fields(match):
    corners = sp.extract_set_pieces(match, kinds=["corner"])
    first = corners[0]
    assert first.team_id == "t_A"
    assert first.player_name == "A. Corner Taker"
    assert first.subtype == "delivery"
    assert first.end_x == 88.0
    assert first.end_y == 45.0


def test_penalty_subtypes(match):
    penalties = sp.extract_set_pieces(match, kinds=["penalty"])
    subtypes = sorted(p.subtype for p in penalties)
    assert subtypes == ["goal", "miss"]


def test_event_index_matches_sorted_events(match):
    events = sp.sorted_events(match)
    set_pieces = sp.extract_set_pieces(match)
    for sp_event in set_pieces:
        raw = events[sp_event.event_index]
        assert raw.get("contestantId") == sp_event.team_id
